### PR TITLE
feat: add decimal size for custom assets

### PR DESCRIPTION
**Pull request by luizstacio**
close: #868

### CODE REVIEW

File: packages/app/src/systems/Core/utils/math.ts [hunk 1]
```diff
@@ -2,8 +2,9 @@ import type { BNInput } from 'fuels';
 import { bn } from 'fuels';
 import { MAX_FRACTION_DIGITS } from '~/config';
 
-export function formatAmount(amount?: BNInput) {
+export function formatAmount(amount?: BNInput, units?: number) {
   return bn(amount).format({
     precision: MAX_FRACTION_DIGITS,
+    units,
```
Comment (LuizAsFight) [inline packages/app/src/systems/Core/utils/math.ts:8]: we don't need this function, can use directly `format` or `formatUnits` from `fuels`

**Comment by LuizAsFight** [comment]
- card looking weird
- we could include decimal information here

image

**Comment by LuizAsFight** [comment]
the amount that I put in doc, doesn't reflect correctly in wallet. I guess need to parse it before sending. (user should know the asset decimal beforehand?

image

**Comment by luizstacio** [comment]
> the amount that I put in doc, doesn't reflect correctly in wallet. I guess need to parse it before sending. (user should know the asset decimal beforehand?
> 
> image

Yeah good catch we probably need to query the assets info on the docs to make the input better. But overall the application also needs to know this info to show the input with the correct format.

File: packages/app/src/systems/Core/utils/math.ts (hunk 1 above)
Comment (luizstacio) [reply packages/app/src/systems/Core/utils/math.ts:8]: Yeah, we should do some refactoring on the code base in the near future.

**Review by LuizAsFight — APPROVED** [review]

**Pull request merged**